diff -ur buildroot-2019.08.orig/board/raspberrypi/genimage-raspberrypi0w.cfg buildroot-2019.08/board/raspberrypi/genimage-raspberrypi0w.cfg
--- buildroot-2019.08.orig/board/raspberrypi/genimage-raspberrypi0w.cfg	2020-04-02 07:55:29.244072244 +0000
+++ buildroot-2019.08/board/raspberrypi/genimage-raspberrypi0w.cfg	2020-04-02 07:57:28.486755936 +0000
@@ -1,7 +1,7 @@
 image boot.vfat {
   vfat {
     files = {
-      "bcm2708-rpi-0-w.dtb",
+      "bcm2708-rpi-zero-w.dtb",
       "rpi-firmware/bootcode.bin",
       "rpi-firmware/cmdline.txt",
       "rpi-firmware/config.txt",
diff -ur buildroot-2019.08.orig/package/gstreamer1/gstreamer1/gstreamer1.mk buildroot-2019.08/package/gstreamer1/gstreamer1/gstreamer1.mk
--- buildroot-2019.08.orig/package/gstreamer1/gstreamer1/gstreamer1.mk	2020-04-02 07:55:29.416070345 +0000
+++ buildroot-2019.08/package/gstreamer1/gstreamer1/gstreamer1.mk	2020-04-02 08:04:58.981783184 +0000
@@ -33,7 +33,6 @@
 	host-pkgconf \
 	libglib2 \
 	$(if $(BR2_PACKAGE_LIBUNWIND),libunwind) \
-	$(if $(BR2_PACKAGE_VALGRIND),valgrind) \
 	$(TARGET_NLS_DEPENDENCIES)
 
 GSTREAMER1_LDFLAGS = $(TARGET_LDFLAGS) $(TARGET_NLS_LIBS)
diff -ur buildroot-2019.08.orig/package/rpi-firmware/rpi-firmware.hash buildroot-2019.08/package/rpi-firmware/rpi-firmware.hash
--- buildroot-2019.08.orig/package/rpi-firmware/rpi-firmware.hash	2020-04-02 07:55:29.812065974 +0000
+++ buildroot-2019.08/package/rpi-firmware/rpi-firmware.hash	2020-04-02 09:13:24.722110230 +0000
@@ -1,3 +1,4 @@
 # Locally computed
 sha256 c82c40cf37fac36160a7f6b9c314506beb942bf113a55d30fa163c56a4f98946 rpi-firmware-bcf40b5c2b94178c7564fb451098d44968e44af5.tar.gz
+sha256 31a84340be08be319570a9d1439f25c0e3513fed73065a56aa5dd61dd605d5d9 rpi-firmware-01ecfd2ba2b7cf3a2f4aa75ada895ee4a3e729f5.tar.gz
 sha256 ba76edfc10a248166d965b8eaf320771c44f4f432d4fce2fd31fd272e7038add boot/LICENCE.broadcom
diff -ur buildroot-2019.08.orig/package/rpi-firmware/rpi-firmware.mk buildroot-2019.08/package/rpi-firmware/rpi-firmware.mk
--- buildroot-2019.08.orig/package/rpi-firmware/rpi-firmware.mk	2020-04-02 07:55:29.812065974 +0000
+++ buildroot-2019.08/package/rpi-firmware/rpi-firmware.mk	2020-04-02 07:56:27.951424179 +0000
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-RPI_FIRMWARE_VERSION = bcf40b5c2b94178c7564fb451098d44968e44af5
+RPI_FIRMWARE_VERSION = 01ecfd2ba2b7cf3a2f4aa75ada895ee4a3e729f5
 RPI_FIRMWARE_SITE = $(call github,raspberrypi,firmware,$(RPI_FIRMWARE_VERSION))
 RPI_FIRMWARE_LICENSE = BSD-3-Clause
 RPI_FIRMWARE_LICENSE_FILES = boot/LICENCE.broadcom
