diff -ur b/package/libplist/libplist.mk a/package/libplist/libplist.mk
--- a/package/libplist/libplist.mk	2021-07-22 07:24:21.600452216 +0000
+++ b/package/libplist/libplist.mk	2021-07-22 07:23:51.256155689 +0000
@@ -16,4 +16,10 @@
 # is not packaged in Buildroot at all.
 LIBPLIST_CONF_OPTS = --without-cython
 
+define LIBPLIST_POST_INSTALL_STAGING
+	cp $(STAGING_DIR)/usr/lib/pkgconfig/libplist-2.0.pc $(STAGING_DIR)/usr/lib/pkgconfig/libplist.pc
+endef
+
+LIBPLIST_POST_INSTALL_STAGING_HOOKS += LIBPLIST_POST_INSTALL_STAGING
+
 $(eval $(autotools-package))
