--- 1/Makefile.am	2023-12-12 09:43:03.108170089 +0000
+++ 2/Makefile.am	2023-12-12 09:43:09.276239530 +0000
@@ -20,7 +20,6 @@
 install-exec-hook:
 if BUILD_FOR_LINUX
 # NQPTP runs as user/group nqptp/nqptp on Linux and uses setcap to access ports 319 and 320
-	setcap 'cap_net_bind_service=+ep' $(bindir)/nqptp
 # no installer for System V
 if INSTALL_SYSTEMD_STARTUP
 	getent group nqptp &>/dev/null || groupadd -r nqptp &>/dev/null
